staging/src/k8s.io/apimachinery/pkg/api/errors/errors_test.go

```diff
@@ -220,3 +220,39 @@ func TestFromObject(t *testing.T) {
 		}
 	}
 }
+
+func TestReasonForErrorSupportsWrappedErrors(t *testing.T) {
+	testCases := []struct {
+		name                string
+		err                 error
+		expectedKnownReason bool
+	}{
+		{
+			name:                "Direct match",
+			err:                 &StatusError{ErrStatus: metav1.Status{Reason: metav1.StatusReasonUnauthorized}},
```

Comment:
I guess that's an example where this would be needed? I was actually curious to know where some of these examples exist.

Reply:
No, the example below ("Nested match") is where this would be needed. I will extend the godocs.